File: design/toolbar/src/main/java/ru/tensor/sbis/design/toolbar/multilinecollapsingtoolbar/ThemeUtils.java
package ru.tensor.sbis.design.toolbar.multilinecollapsingtoolbar;

import android.content.Context;
import android.content.res.TypedArray;

/**
 * Утилиты для работы с темами
 *
 * @author us.bessonov
 */
class ThemeUtils {

    private static final int[] APPCOMPAT_CHECK_ATTRS = {
            com.google.android.material.R.attr.colorPrimary
    };

    static void checkAppCompatTheme(Context context) {
        TypedArray a = context.obtainStyledAttributes(APPCOMPAT_CHECK_ATTRS);
        final boolean failed = !a.hasValue(0);
        a.recycle();
        if (failed) {
            throw new IllegalArgumentException("You need to use a Theme.AppCompat theme "
                    + "(or descendant) with the design library.");
        }
    }
}

File: common/app_file_browser/src/main/java/ru/tensor/sbis/app_file_browser/util/ThemeUtils.kt
package ru.tensor.sbis.app_file_browser.util

import android.content.Context
import android.view.ContextThemeWrapper
import ru.tensor.sbis.app_file_browser.R
import ru.tensor.sbis.design.utils.getDataFromAttrOrNull

/**
 * Возвращает [ContextThemeWrapper] с темой из атрибута [R.attr.appFileBrowserTheme], если она задана.
 */
internal fun getThemedContext(context: Context): Context {
    val theme = context.getDataFromAttrOrNull(R.attr.appFileBrowserTheme)
        ?: tryGetFileBrowserThemeFromAppTheme(context)
    return theme?.let { ContextThemeWrapper(context, it) }
        ?: context
}

private fun tryGetFileBrowserThemeFromAppTheme(context: Context): Int? =
    ContextThemeWrapper(context.applicationContext, context.applicationInfo.theme)
        .getDataFromAttrOrNull(R.attr.appFileBrowserTheme)